src/main/webapp/progress.js

```diff
@@ -15,18 +15,15 @@
 /**
  * Function for progess.html, checks what kind of user the viewer is, and loads progress information
  */
-function loadProgress() {
-    fetch('/login-status').then(response => response.json()).then((loginStatus) => {
-        console.log(loginStatus);    
-        loadProgressHelper(document, loginStatus);
-    });
-}
+function loadProgress(document, loginStatus, user) {
+    addEventListeners();
 
-function loadProgressHelper(document, loginStatus) {
-    getExperiences(document, loginStatus);
-    getGoals(document, loginStatus);
+    document.getElementById('progress-tracker').style.display = 'block';
+
+    getExperiences(document, loginStatus, user);
+    getGoals(document, loginStatus, user);
     getAchievements(document, loginStatus);
-    getPastSessionsAndTopics(document, loginStatus);
+    getPastSessionsAndTopics(document, loginStatus, user);
```

Comment:
I think I missed this but when/where do we get the `user` from?

Reply:
The user is retrieved when the displayProfile function is called and then from there it is passed down to the function that need it.